src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -0,0 +1,416 @@
+                case EVENT_MORLEN_2:
+                    MorlenTalk(SAY_MORLEN_5);
+
+                    _events.ScheduleEvent(EVENT_MORLEN_ATTACK, 1s);
+                    break;
+                case EVENT_MORLEN_ATTACK:
+                    if (Creature* morlen = ObjectAccessor::GetCreature(*me, _morlenGUID))
+                    {
+                        morlen->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC | UNIT_FLAG_IMMUNE_TO_NPC);
+                        if (Creature* thalorien = ObjectAccessor::GetCreature(*me, _thalorienGUID))
+                            morlen->AI()->AttackStart(thalorien);
+                    }
+
+                    break;
+                case EVENT_OUTRO_1:
+                    if (Creature* thalorien = ObjectAccessor::GetCreature(*me, _thalorienGUID))
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                            thalorien->AI()->DoCast(player, SPELL_KILL_CREDIT);
+
+                    ThalorienTalk(SAY_THALORIEN_7);
+
+                    _events.ScheduleEvent(EVENT_OUTRO_2, 5s);
+                    break;
+                case EVENT_OUTRO_2:
+                    ThalorienTalk(SAY_THALORIEN_8);
+
+                    _events.ScheduleEvent(EVENT_OUTRO_3, 7s);
+                    break;
+                case EVENT_OUTRO_3:
+                    if (Creature* thalorien = ObjectAccessor::GetCreature(*me, _thalorienGUID))
+                        thalorien->AI()->DoCastSelf(SPELL_POLYMORPH_VISUAL);
+
+                    ThalorienTalk(SAY_THALORIEN_9);
+
+                    _events.ScheduleEvent(EVENT_OUTRO_4, 5s);
+                    break;
+                case EVENT_OUTRO_4:
+                    ThalorienTalk(SAY_THALORIEN_10);
+
+                    _events.ScheduleEvent(EVENT_KNEEL, 6s);
+                    break;
+                case EVENT_KNEEL:
+                    if (Creature* thalorien = ObjectAccessor::GetCreature(*me, _thalorienGUID))
+                    {
+                        thalorien->SetStandState(UNIT_STAND_STATE_KNEEL);
+                        thalorien->DespawnOrUnsummon(5 * IN_MILLISECONDS);
+                    }
+
+                    me->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+
+                    break;
+            }
+        }
+    }
+
+    void JustSummoned(Creature* summon) override
+    {
+        _summons.Summon(summon);
+    }
+
+    void SummonedCreatureDespawn(Creature* summon) override
+    {
+        _summons.Despawn(summon);
+
+        if (_summons.size() == 2)
+        {
+            if (summon->GetEntry() == NPC_SCOURGE_ZOMBIE)
+                _events.ScheduleEvent(EVENT_SPAWN_WAVE_2, 1s);
+            else if (summon->GetEntry() == NPC_GHOUL_INVADER)
+                _events.ScheduleEvent(EVENT_SPAWN_WAVE_3, 1s);
+            else if (summon->GetEntry() == NPC_CRYPT_RAIDER)
+                _events.ScheduleEvent(EVENT_MORLEN_2, 1s);
+        } else if (_summons.size() == 1)
```

Comment:
we don't usually have "}" on the same line as "else", please move the "else" on its own line